### PR TITLE
feat(hosting.database): replace deprecated routes for ram upgrade

### PR DESCRIPTION
**Pull request by lucas-chaigne**
| Question         | Answer
| ---------------- | ---
| Branch?          | develop
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #WEB-15524
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

exact same code as https://github.com/ovh/manager/pull/9694
during testing phase this PR was reverted because of a bug found ( err 500 on https://api.ovh.com/console/#/order/upgrade/cloudDB/%7Bdomain%7D~GET ).
We couldnt replicate the 500 err neither via apiV6 call or via staging link, with the exact same service in the same state.
There may have been a fix since on agora/db/iceberg... side but it's hard to know

### CODE REVIEW

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js [hunk 1]
```diff
@@ -115,19 +117,25 @@ angular.module('App').controller(
...
-        );
+        .getRamPlan(this.productId, this.model.ram.planCode)
+        .then((order) => {
+          this.loading.durations = false;
+
+          const details = order.details.filter(({ detailType }) => {
+            return detailType === 'DURATION';
+          });
+
+          const { description } = details[0];
```
Comment (qpavy) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:128]:
```suggestion
          const [{ description }] = details;
```

**Review by qpavy — APPROVED** [review]

File: packages/manager/apps/web/client/app/private-database/private-database.service.js [hunk 2]
```diff
@@ -485,62 +484,24 @@ export default class PrivateDatabase {
...
+    return this.$http
+      .get(`${this.swsProxypassUpgradePath}/${serviceName}/${planCode}`, {
+        params: {
+          quantity: 1,
+        },
       })
-      .then(() => defer.resolve(durationsTab))
-      .catch(() => defer.resolve(durationsTab));
-    return defer.promise;
+      .then((response) => response.data.order);
```
Comment (darsene) [inline packages/manager/apps/web/client/app/private-database/private-database.service.js:494]:
```suggestion
      .then(({data: {order}}) => order);
```
use destructed object instead

File: packages/manager/apps/web/client/app/private-database/private-database.service.js [hunk 3]
```diff
@@ -485,62 +484,24 @@ export default class PrivateDatabase {
...
     return this.$http
-      .post(`${this.swsProxypassOrderPath}/${serviceName}/ram/${duration}`, {
-        ram,
+      .post(`${this.swsProxypassUpgradePath}/${serviceName}/${planCode}`, {
+        params: {
+          quantity: 1,
+        },
       })
-      .then((response) => response.data);
+      .then((response) => response.data.order);
```
Comment (darsene) [inline packages/manager/apps/web/client/app/private-database/private-database.service.js:504]:
```suggestion
     .then(({data: {order}}) => order);
```

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js [hunk 4]
```diff
@@ -40,38 +39,41 @@ angular.module('App').controller(
...
-          if (this.database.infrastructure === 'legacy') {
-            remove(this.data.availableRam, (ram) => ram === '2048');
-          }
-
-          remove(
-            this.data.availableRam,
-            (ram) => +ram === +this.database.ram.value,
-          );
+        .then((results) => {
+          results.forEach((plan) => {
```
Comment (darsene) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:63]:
```suggestion
          results.forEach(({planCode}) => {
```

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js [hunk 5]
```diff
@@ -40,38 +39,41 @@ angular.module('App').controller(
...
+            const capacity = parseInt(plan.planCode.match(/[0-9]+/)?.[0], 10);
```
Comment (darsene) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:64]:
```suggestion
            const capacity = parseInt(planCode.match(/[0-9]+/)?.[0], 10);
```

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js [hunk 6]
```diff
@@ -40,38 +39,41 @@ angular.module('App').controller(
...
+
+            if (
+              this.database.infrastructure !== 'legacy' ||
+              capacity !== 2048 ||
+              +capacity !== +this.database.ram.value
+            ) {
+              this.data.availableRam.push({
+                capacity,
+                planCode: plan.planCode,
```
Comment (darsene) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:73]:
```suggestion
                planCode,
```

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js (hunk 1 above)
Comment (darsene) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:128]:
```suggestion
          const  { description } = details.find(({ detailType }) =>
             detailType === 'DURATION';
          );
```

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js [hunk 7]
```diff
@@ -115,19 +117,25 @@ angular.module('App').controller(
...
-            this.loading.durations = false;
-            this.data.durations.available = durations;
-          },
-          angular.noop,
-          (duration) => {
-            this.data.durations.available = duration;
-          },
-        );
+        .getRamPlan(this.productId, this.model.ram.planCode)
+        .then((order) => {
```
Comment (darsene) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:121]:
```suggestion
        .then(( {details, contracts, prices }) => {
```

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js [hunk 8]
```diff
@@ -115,19 +117,25 @@ angular.module('App').controller(
...
+          });
+
+          const { description } = details[0];
+
+          this.data.durations.available = [
+            {
+              prices: order.prices,
+              description,
+              details: order.details,
+              contracts: order.contracts,
```
Comment (darsene) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:135]:
```suggestion
              prices,
              description,
              details,
              contracts,
```

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js (hunk 1 above)
Comment (darsene) [reply packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:128]: Should be simplified with `find` as only the first element is used

**Review by darsene — CHANGES_REQUESTED** [review]

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js (hunk 4 above)
Comment (lucas-chaigne) [reply packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:63]: seen together (as well as other comments suggesting de-structuring) : linter accepts both

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js (hunk 1 above)
Comment (lucas-chaigne) [reply packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js]: done

**Review by darsene — DISMISSED** [review]

**Review by anooparveti — APPROVED** [review]

**Comment by anooparveti** [comment]
Rebasing with `release/components-w42`

**Review by anooparveti — APPROVED** [review]

**Review by darsene — APPROVED** [review]

**Pull request merged**